### PR TITLE
Fix math_builtin_api tests

### PR DESCRIPTION
**Pull request by aleksmesh**
Fixed bugs in python module and math functions for marray in util.

### CODE REVIEW

File: tests/math_builtin_api/math_builtin.h [hunk 1]
```diff
@@ -130,16 +131,28 @@ bool verify(sycl_cts::util::logger &log, sycl::vec<T, N> a,
...
-            std::string comment) {
+            const std::string &comment) {
   return verify(log, a, sycl_cts::resultRef<T>(b), accuracy, comment);
 }
 
 template <int N, typename returnT, typename funT>
 void check_function(sycl_cts::util::logger &log, funT fun,
                     sycl_cts::resultRef<returnT> ref, int accuracy = 0,
-                    std::string comment = "") {
+                    const std::string &comment = "") {
```
Comment (keryell) [inline tests/math_builtin_api/math_builtin.h:155]: While you are here :-)
```suggestion
                    const std::string &comment = {}) {
```
You do not want every time to measure this empty C string and then construct a `std::string` from it...

File: tests/math_builtin_api/math_builtin.h [hunk 2]
```diff
@@ -166,9 +179,8 @@ void check_function(sycl_cts::util::logger &log, funT fun,
 template <int N, typename returnT, typename funT, typename argT>
 void check_function_multi_ptr_private(sycl_cts::util::logger &log, funT fun,
                                       sycl_cts::resultRef<returnT> ref,
-                                      argT ptrRef,
-                                      int accuracy = 0,
-                                      std::string comment = "") {
+                                      argT ptrRef, int accuracy = 0,
+                                      const std::string &comment = "") {
```
Comment (keryell) [inline tests/math_builtin_api/math_builtin.h:183]:
```suggestion
                                      const std::string &comment = {) {
```
everywhere.

Comment (aleksmesh) [reply tests/math_builtin_api/math_builtin.h]: Fixed it. Thanks!

File: tests/math_builtin_api/math_builtin.h (hunk 1 above)
Comment (aleksmesh) [reply tests/math_builtin_api/math_builtin.h]: Done.

File: tests/math_builtin_api/math_builtin.h [hunk 3]
```diff
@@ -102,15 +102,15 @@ verify(sycl_cts::util::logger &log, T value, sycl_cts::resultRef<T> r, int accur
...
-    msg += ", " + comment;
+  if (!comment.empty()) msg += ", " + comment;
   log.note(msg);
   return false;
 }
 
 template <typename T>
-typename std::enable_if<std::is_integral<T>::value, bool>::type
-verify(sycl_cts::util::logger &log, T value, sycl_cts::resultRef<T> r, int, std::string) {
+typename std::enable_if<std::is_integral<T>::value, bool>::type verify(
```
Comment (keryell) [inline tests/math_builtin_api/math_builtin.h:111]: Since you are here :-)
```suggestion
typename std::enable_if_t<std::is_integral_v<T>, bool> verify(
```

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**